src/ray/core_worker/core_worker.cc

```diff
@@ -1895,12 +1895,31 @@ void CoreWorker::HandlePushTask(const rpc::PushTaskRequest &request,
     return;
   }
 
+  const TaskSpecification task_spec(request.task_spec());
```

Comment:
Could we avoid doing this copy? I believe you can check whether the task is an actor task by checking `request.task_spec().type() == TaskType::ACTOR_TASK`.